examples/nextjs-blog/pages/blog/post.js

```diff
@@ -1,38 +1,89 @@
-import React from 'react'
+import React, { PureComponent } from 'react'
 import Markdown from 'markdown-to-jsx'
 import PropTypes from 'prop-types'
 import { flamelinkApp as app } from '../../utils/flamelink'
-import { getPostsWithMedia, getImageAlt } from '../../utils/post/post.util'
-
-const Post = function(props = {}) {
-  const { post } = props
-
-  if (post) {
-    const {
-      title,
-      date,
-      author: { displayName },
-      content,
-      imageURL,
-    } = post
-
-    return (
-      <div>
-        <h1>{title}</h1>
-        <h2>{displayName}</h2>
-        <p>{new Date(date).toLocaleDateString()}</p>
-        <Markdown>{content}</Markdown>
-        <img src={imageURL} alt={getImageAlt(post)} />
-      </div>
-    )
+import {
+  getPostsWithMedia,
+  getImageAlt,
+  getDateString,
+} from '../../utils/post/post.util'
+
+class Post extends PureComponent {
+  constructor(props) {
+    super(props)
+    this._subscription = null
+    this.state = {
+      post: null,
+    }
   }
 
-  return 'No post data!'
+  componentDidMount() {
+    // todo: this isn't so lekker because the path may change in future. is it
+    // possible to share the `query.slug` value from the server's execution with
+    // the client?
+    const { pathname } = location
+    const slug = pathname.replace('/blog/', '')
```

Comment:
maybe split the `pathname` by `/` and pop off the last item in the array as the `slug`?